### PR TITLE
empirical disp updates for qcng v0.19

### PR DESCRIPTION
**Pull request by loriab**
## Description
It seems that in the course of harmonizing dftd4 in QCEngine, I touched enough dftd3, gcp, and mp2d that #2142 (dftd4 and minor dftd3, gcp, mp2d) can't be put off after the QCEngine v0.19 release. So here's the non-dftd4 tweaks.

## Todos
- [x] switch gcp interface to QCEngine
- [x] update types return for disp. engine can handle ndarray, so leave arrays as np and shaped rather than flat lists
- [x] Note that this won't build until new QCEngine release

## Checklist
- [x] Tests added for any new features
- [ ] [All or relevant fraction of full tests run](http://psicode.org/psi4manual/master/build_planning.html#how-to-run-a-subset-of-tests)

## Status
- [x] Ready for review
- [ ] Ready for merge

### CODE REVIEW

File: psi4/driver/qcdb/molecule.py
Comment (JonathonMisiewicz) [inline psi4/driver/qcdb/molecule.py:1364]: What are the allowed ranges?

File: psi4/driver/qcdb/molecule.py
Comment (JonathonMisiewicz) [inline psi4/driver/qcdb/molecule.py:1398]: The intention of the code isn't clear, so it took me a minute to understand this comment. I think this is trying to reshape the gradient to number atoms * 3?

File: psi4/driver/qcdb/molecule.py
Comment (JonathonMisiewicz) [inline psi4/driver/qcdb/molecule.py:1408]: Can we rename `k` and `qca` to `variable_name` and `value`, or something similar? This should happen above as well. It's more readable for those of us who aren't in-tune with QCEngine conventions.

**Review by JonathonMisiewicz — APPROVED** [review]
I have some minor clarity suggestions, but LGTM otherwise, once 0.19 is in.

**Review by hokru — APPROVED** [review]

File: psi4/driver/qcdb/molecule.py
Comment (loriab) [inline psi4/driver/qcdb/molecule.py:1397]:
```suggestion
        # hack (instead of checking dertype GRAD) to collect `(nat, 3)` ndarray of gradient if present
```

Comment (loriab) [reply psi4/driver/qcdb/molecule.py:1397]: right, gradient could be present as a flat list

File: psi4/driver/qcdb/molecule.py
Comment (loriab) [inline psi4/driver/qcdb/molecule.py:1409]:
```suggestion
            for variable_name, qcv in jobrec['extras']['qcvars'].items():
                if not isinstance(qcv, (list, np.ndarray)):
                    core.set_variable(variable_name, float(qcv))
```

File: psi4/driver/qcdb/molecule.py
Comment (loriab) [inline psi4/driver/qcdb/molecule.py:1400]:
```suggestion
        for variable_name, qcv in jobrec['extras']['qcvars'].items():
            if isinstance(qcv, (list, np.ndarray)):
                jobrec['extras']['qcvars'][variable_name] = np.array(qcv).reshape(-1, 3)
```

**Review by jturney — APPROVED** [review]

File: psi4/driver/qcdb/molecule.py
Comment (loriab) [inline psi4/driver/qcdb/molecule.py:1339]:
```suggestion
    def run_gcp(self, func: str = None, dertype: Union[int, str] = None, verbose: int = 1):
```

File: psi4/driver/qcdb/molecule.py
Comment (loriab) [inline psi4/driver/qcdb/molecule.py:1364]:
```suggestion
            Amount of printing. Unused at present.
```

**Pull request merged**
